src/Build/BackEnd/Client/MSBuildClient.cs

```diff
@@ -207,6 +218,12 @@ public MSBuildClientExitResult Execute(CancellationToken cancellationToken)
 
             MSBuildEventSource.Log.MSBuildServerBuildStop(descriptiveCommandLine, _numConsoleWritePackets, _sizeOfConsoleWritePackets, _exitResult.MSBuildClientExitType.ToString(), _exitResult.MSBuildAppExitTypeString);
             CommunicationsUtilities.Trace("Build finished.");
+
+            if (_originalConsoleMode is not null)
+            {
+                NativeMethodsShared.SetConsoleMode(stdOut, _originalConsoleMode.Value);
```

Comment:
Since identical cleanup code is required for multiple exit code paths should it be in a try/finally?